Server dev

nks5295 opened the pull request:
Handler code is good to go:
- relies on marshal code to work
  Server should be fine
- now initializes gwkv_server struct
- now relies on handler for message parsing

nks5295 committed:
Adding the start of handle_operation.c/h.

The basic structure of the function is set up

The get switch case is almost done

Need to finish set case and then the sending over socket

nks5295 committed:
Merge branch 'master' into handler

phil-lopreiato committed:
Update Makefile structure for passing tests

Conflicts:
	lib/marshal/Makefile

nks5295 committed:
handle_operation parses the message received and then crafts the message
to send based on the method type.

It is ALMOST DONE -- the only thing left to do is add the calls to send
the message over the sockfd

nks5295 committed:
Merge branch 'master' into handler

nks5295 committed:
Merge branch 'handler' into server-dev

Conflicts:
	lib/marshal/Makefile

nks5295 committed:
Modified handle_operation files to reflect changes necessary to add ht
struct inside the server-kv code

nks5295 committed:
No need for sockfd anywhere in handler so removed it

nks5295 committed:
Removed misc functions for parsing since they are not done in server.
Added gwkv_server struct init code in server main function.
Redid message parsing in server thread_handler function

phil-lopreiato commented:
Tests failed, see https://travis-ci.org/gwAdvNet2015/gw-kv-store/builds/59294121

```
make[1]: Entering directory `/home/travis/build/gwAdvNet2015/gw-kv-store/server'
gcc -pthread -c server-kv.c server-kv-run.c
In file included from server-kv.c:14:0:
handle_operation.h:10:28: fatal error: ../lib/marshal.h: No such file or directory
compilation terminated.
make[1]: *** [build-srcs] Error 1
make[1]: Leaving directory `/home/travis/build/gwAdvNet2015/gw-kv-store/server'
make: *** [server] Error 2
```

Looks like you have to fix a header location

nks5295 committed:
Merge branch 'server-dev' of github.com:gwAdvNet2015/gw-kv-store into server-dev

nks5295 committed:
Removing method enum, operation struct definition, and return status macro defines since they have previously been moved to marshal.h

nks5295 committed:
Fixing include for lib/marshal/marshal.h

nks5295 committed:
Modified server files to handle hashtable handler functions

nks5295 committed:
Adding include to lib/marshal/marshal.h since that's where the status
codes are now

nks5295 committed:
code cleanup -- removed unnecessary comments

nks5295 committed:
Changing return type of function gwkv_handle_operation

amygara commented:
@nks5295 Good progress, Travis fails because you need to link in your handle_operation objects, see server/test/Makefile for example. Almost ready to merge.

amygara commented on server/server-kv.h:
```diff
@@ -1,27 +1,13 @@
-//void parse_message(char *message, int len, struct operation *operation);
-//void process_operation(struct operation *operation, char *message, int *len);
+// This is the hashtable on the server
+struct gwkv_server *ht;
```
Why is this here and not in server-kv.c? I may just be being a dumb right now, but my gut says it should be in the .c.

nks5295 committed:
Adding changes to makefile based on previous.. (I have no way to test on
my comptuer since my linker remains busted, so im depending on TravisCI
for debugging my changes here)
